src/main/java/uk/gov/laa/ccms/caab/controller/requests/ProviderRequestsController.java

```diff
@@ -157,16 +160,23 @@ public String requestTypePost(
 
   /**
    * Populates dropdown options for provider request types form.
+   * also filters based on user function codes
    *
    * @param model The model for the view.
+   * @param userDetail Logged-in user details.
    */
-  protected void populateProviderRequestTypes(final Model model) {
+  protected void populateProviderRequestTypes(final Model model, UserDetail userDetail) {
+
+    List<String> functions = userDetail.getFunctions();
     final List<ProviderRequestTypeLookupValueDetail> providerRequestTypes = Optional.ofNullable(
-            lookupService.getProviderRequestTypes(false, null).block())
-        .map(ProviderRequestTypeLookupDetail::getContent)
+            lookupService.getProviderRequestTypes(false, null)
+                .map(ProviderRequestTypeLookupDetail::getContent)
+                .flatMapMany(Flux::fromIterable)
+                .filter(it -> it.getAccessFunctionCode() == null
+                    || functions == null || functions.contains(it.getAccessFunctionCode()))
```

Comment:
`functions == null` suggests users without any permissions have access to the complete list? This doesn't seem right. Does it behave the same way in old PUI?

```suggestion
                    || functions.contains(it.getAccessFunctionCode()))
```